source/includes/fundamentals/code-snippets/builders/AggBuilders.java

```diff
@@ -0,0 +1,287 @@
+                new Document("_id", 3).append("stock_item", "almonds").append("warehouse", "B").append("instock", 60),
+                new Document("_id", 4).append("stock_item", "cookies").append("warehouse", "B").append("instock", 40),
+                new Document("_id", 5).append("stock_item", "cookies").append("warehouse", "A").append("instock", 80)));
+        collection = database.getCollection("orders");
+        collection.drop();
+        collection.insertMany(
+                asList(new Document("_id", 1).append("item", "almonds").append("price", 12).append("ordered", 2),
+                        new Document("_id", 2).append("item", "pecans").append("price", 20).append("ordered", 1),
+                        new Document("_id", 3).append("item", "cookies").append("price", 10).append("ordered", 60)));
+        // begin advanced lookup
+        List<Variable<String>> variables = asList(new Variable<>("order_item", "$item"),
+                new Variable<>("order_qty", "$ordered"));
+
+        List<Bson> pipeline = asList(
+                match(Filters.expr(new Document("$and",
+                        asList(new Document("$eq", asList("$$order_item", "$stock_item")),
+                                new Document("$gte", asList("$instock", "$$order_qty")))))),
+                project(Projections.fields(Projections.exclude("stock_item"), Projections.excludeId())));
+
+        MongoCursor<Document> cursor = collection
+                .aggregate(asList(lookup("warehouses", variables, pipeline, "stockdata"))).cursor();
+        // end advanced lookup
+        cursor.forEachRemaining(doc -> System.out.println(doc.toJson()));
+        database = mongoClient.getDatabase("sample_mflix");
+        collection = database.getCollection("movies");
+    }
+
+    private void basicLookupStage() {
+        // begin basic lookup
+        lookup("comments", "_id", "movie_id", "joined_comments");
+        // end basic lookup
+    }
+
+    private void limitStage() {
+        // begin limit
+        limit(10);
+        // end limit
+    }
+
+    private void skipStage() {
+        // begin skip
+        skip(5);
+        // end skip
+    }
+
+    private void sortStage() {
+        // begin sortStage
+        sort(orderBy(descending("year"), ascending("title")));
+        // end sortStage
+    }
+
+    private void sampleStage() {
+        // begin sample
+        sample(5);
+        // end sample
+    }
+
+    private void projectComputed() {
+        // begin computed
+        project(Projections.fields(Projections.computed("rating", "$rated"), Projections.excludeId()));
+        // end computed
+    }
+
+    private void projectStage() {
+        // begin project
+        project(Projections.fields(Projections.include("title", "plot"), Projections.excludeId()));
+        // end project
+    }
+
+    private void matchStage() {
+        // begin match
+        match(Filters.eq("title", "The Shawshank Redemption"));
```

Comment:
It's confusing if the code imports Filters statically, but doesn't actually use the import.  Let's remove all the class names, e.g.:

    match(eq("title", "The Shawshank Redemption"));